crates/weaver_codegen_test/build.rs

```diff
@@ -0,0 +1,219 @@
+    let schema = SchemaResolver::resolve_semantic_convention_registry(&mut registry)
+        .unwrap_or_else(|e| process_error(&logger, e));
+
+    let config = GeneratorConfig::new(TEMPLATES_PATH.into());
+    let engine = TemplateEngine::try_new(&format!("registry/{}", TARGET), config)
+        .unwrap_or_else(|e| process_error(&logger, e));
+    let template_registry = TemplateRegistry::try_from_resolved_registry(
+        schema
+            .registry(REGISTRY_ID)
+            .expect("Failed to get the registry from the resolved schema"),
+        schema.catalog(),
+    )
+    .unwrap_or_else(|e| process_error(&logger, e));
+    let target_dir: PathBuf = target_dir.into();
+    engine
+        .generate(logger.clone(), &template_registry, target_dir.as_path())
+        .unwrap_or_else(|e| process_error(&logger, e));
+
+    print_logs(&logger);
+
+    // For the purpose of the integration test located in `tests/codegen.rs`, we need to:
+    // - Combine all generated files to form a single file containing all the generated code
+    // organized in nested modules.
+    // - Replace `//!` with `//`
+    // - Remove `pub mod` lines
+    create_single_generated_rs_file(target_dir.as_path());
+}
+
+/// Create a single generated.rs file containing all the generated code organized in nested modules.
+fn create_single_generated_rs_file(root: &Path) {
+    let mut root_module = Module {
+        name: "".to_owned(),
+        content: "".to_owned(),
+        sub_modules: HashMap::new(),
+    };
+
+    // Traverse the directory and add the content of each file to the root module
+    for entry in walkdir::WalkDir::new(root) {
+        let entry = entry.expect("Failed to read entry");
+        let path = entry.path();
+
+        // Only process files with the .rs extension that contain the generated comment
+        if path.is_file() && path.extension().map_or(false, |ext| ext == "rs") {
+            let file_content = std::fs::read_to_string(path).expect("Failed to read file");
+
+            // Only process files that have been generated
+            // This test expects the comment `DO NOT EDIT, THIS FILE HAS BEEN GENERATED BY WEAVER`
+            // to be present in each generated file.
+            if file_content.contains("DO NOT EDIT, THIS FILE HAS BEEN GENERATED BY WEAVER") {
+                let relative_path = path.strip_prefix(root).expect("Failed to strip prefix");
+                let file_name = path
+                    .file_stem()
+                    .expect("Failed to extract the file name")
+                    .to_str()
+                    .expect("Failed to convert to string");
+                let parent_modules = relative_path.parent().map_or(vec![], |parent| {
+                    let parent = parent.display().to_string();
+                    if parent.is_empty() {
+                        vec![]
+                    } else {
+                        parent.split('/').map(|s| s.to_owned()).collect::<Vec<_>>()
```

Comment:
This line may not be windows friendly.  I think you should be able to split using `Path` in some fashion.

Reply:
Fixed. Path.components() is the platform independent solution.

Reply:
Well not completely...